gui/mozregui/addons_editor.py

```diff
@@ -0,0 +1,85 @@
+from PyQt4.QtCore import QAbstractListModel, QModelIndex, Qt, \
+    pyqtSlot as Slot
+from PyQt4.QtGui import QWidget, QFileDialog
+from mozregui.ui.addons_editor import Ui_AddonsEditor
+
+
+class AddonsMode(QAbstractListModel):
+    """
+    A Qt model that can edit addons path.
+    """
+
+    def __init__(self, parent=None):
+        QAbstractListModel.__init__(self, parent)
+        self.addonsPath = []
+
+    def rowCount(self, index=QModelIndex()):
+        return len(self.addonsPath)
+
+    def data(self, index, role=Qt.DisplayRole):
+        if role == Qt.DisplayRole:
+            return self.addonsPath[index.row()]
+
+    def headerData(self, section, orientation, role=Qt.DisplayRole):
+        if role == Qt.DisplayRole and orientation == Qt.Horizontal:
+            return 'path'[section]
+
+    def flags(self, index):
+        return Qt.ItemIsSelectable | Qt.ItemIsEditable | Qt.ItemIsEnabled
+
+    def add_addons_from_file(self, addons):
+        if addons:
+            addons_list_length = len(self.addonsPath)
+            self.beginInsertRows(QModelIndex(), addons_list_length,
+                                 addons_list_length)
+            self.addonsPath.append(str(addons))
+            self.endInsertRows()
+
+    def remove_pref(self, row):
+        self.beginRemoveRows(QModelIndex(), row)
+        self.addonsPath.pop(row)
+        self.endRemoveRows()
+
+
+class AddonsWidgetEditor(QWidget):
+    """
+    A widget to add or remove addons, and buttons to let the user interact.
+    """
+    def __init__(self, parent=None):
+        QWidget.__init__(self, parent)
+        self.ui = Ui_AddonsEditor()
+        self.ui.setupUi(self)
+
+        self.list_model = AddonsMode()
+        self.ui.list_view.setModel(self.list_model)
+
+    @Slot()
+    def add_addons_from_file(self):
+        fname = QFileDialog.getOpenFileName(
```

Comment:
you could use http://doc.qt.io/qt-4.8/qfiledialog.html#getOpenFileNames, so the user can choose multiple addons at the same time